pontoon/base/models.py

```diff
@@ -25,12 +25,101 @@
+        with counts of translations.
+        Method supports exclusion of fields, you can exclude field if param contains
+        exclude__ prefix
+        e.g.
+        _changed_translations_counts(exclude__approved=True)
+        Will exclude "approved" field with True value.
+        """
+        translation_query = (
+            ~Q(translation__string=F('translation__entity__string'))
+            & ~Q(translation__string=F('translation__entity__string'))
+        )
+
+        if filter:
+            translation_query &= Q(**filter)
+
+        if exclude:
+            translation_query &= ~Q(**exclude)
+
+        if start_date and end_date:
+            translation_query &= Q(translation__date__range=(start_date, end_date))
+
+        # For each translation that matches the filter, return 1. Aggregate
+        # the sum of all those results to count the number of matching
+        # translations.
+        return Sum(
+                Case(
+                    When(translation_query, then=1),
+                    output_field=models.IntegerField(),
+                    default=0
+                    )
+                )
+
+    def get_changed_translations(self, start_date=None, end_date=None):
+        translations = (
+            Translation.objects
+            .exclude(user=None, string=F('entity__string'))
+            .exclude(string=F('entity__string_plural'))
+        )
+        if start_date and end_date:
+            translations = translations.filter(date__range=(start_date, end_date))
+        return translations
+
+    def get_approved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(approved=True)
+
+    def get_unapproved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).exclude(approved=True, fuzzy=True)
+
+    def get_needs_work(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(fuzzy=True)
+
+    def get_contributors_with_stats(self, start_date=None, end_date=None, limit=100):
+        """
+        Returns contributors list, sorted by count of their translations.
+        Every user instance has added following properties:
+        * translations_count
+        * translations_approved_count
+        * translations_unapproved_count
+        * translations_needs_work_count
+        Method has been created mainly to improve performance and to optimize
+        count of sql queries during generation of metrics.
+        """
+        def translations_counts(filter=None, exclude=None):
+            """Short helper to avoid duplication of passing dates."""
+            return self._changed_translations_counts(start_date, end_date, filter=filter, exclude=exclude)
+
+        contributors = (
+            self
+            .filter(translation__in=self.get_changed_translations(start_date, end_date)).distinct()
```

Comment:
This is unnecessary because the filters in `_changed_translations_counts` already filter out unchanged translations. You can remove this line and the entire `get_changed_translations` method.